### PR TITLE
ExperienceMechanic implemented

### PR DESCRIPTION
**Pull request by neYaANK**
#405 Implementation

### CODE REVIEW

File: src/main/java/com/sucy/skill/dynamic/mechanic/ExperienceMechanic.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        PlayerClass playerClass = SkillAPI.getPlayerData(player).getClass(group);
+        double allNextLevelExp  = playerClass.getData().getRequiredExp(playerClass.getLevel());
+        double amount           = 0;
+        if(flat)
+            amount = expValue;
+        else if(percent)
+            amount = allNextLevelExp*expValue/100;
+        switch (mode){
+            case "give":
+                playerClass.giveExp(amount, ExpSource.SPECIAL);
```
Comment (neYaANK) [inline src/main/java/com/sucy/skill/dynamic/mechanic/ExperienceMechanic.java:56]: Here i am not sure about the proper `ExpSource` to use. Thought of using `ExpSource.PLUGIN` but that didn't seem to work for me

Comment (Travja) [reply src/main/java/com/sucy/skill/dynamic/mechanic/ExperienceMechanic.java:56]: PLUGIN should be a valid xp source, and probably would be appropriate here. It just didn't work?

Comment (neYaANK) [reply src/main/java/com/sucy/skill/dynamic/mechanic/ExperienceMechanic.java:56]: Just checked it one more time, yes it does not work as this `ExpSource` is not considered valid. If it is an error, I guess I can then modify the 'class receives from source' check to accept `PLUGIN` regardless of `ExpSource` settings

**Comment by Travja** [comment]
Can you add this to the new editor by creating a PR on https://github.com/Travja/editor

File: src/main/java/com/sucy/skill/listener/ExperienceListener.java [hunk 2]
```diff
@@ -86,6 +91,17 @@ public void onCraft(CraftItemEvent event) {
...
+    @EventHandler(priority = EventPriority.LOWEST)
+    public void onExperienceGain(PlayerExperienceGainEvent event) {
+        Player player = event.getPlayerData().getPlayer();
+        if (event.isCancelled()
+                && event.getSource() == ExpSource.PLUGIN
+                && exempt.containsKey(player.getUniqueId())
+                && exempt.get(player.getUniqueId()) == event.getExp()) {
+            event.setCancelled(false);
+        }
+    }
```
Comment (Travja) [inline src/main/java/com/sucy/skill/listener/ExperienceListener.java:103]: @neYaANK I wanted to set up a way to still let the class settings take effect, while still allowing the exp to go through, so this is what I came up with. I wasn't able to test it, though. So if you wouldn't mind making sure that this works as I expect it to, that would be awesome :D

Comment (neYaANK) [reply src/main/java/com/sucy/skill/listener/ExperienceListener.java:103]: It does work, but it's worth mentioning that it requires `exp.yml`'s `enabled` field to be `true` which is by default `false` and that may cause some confusion

Comment (Travja) [reply src/main/java/com/sucy/skill/listener/ExperienceListener.java:103]: Oh does it? It doesn't even fire off the event if it is false?

**Pull request merged**